iscript/src/iscript/macvpn.py

```diff
@@ -305,7 +296,40 @@ async def vpn_behavior(config, task, notarize=True):
         provisionprofile_filename="orgmozillamacosFirefoxVPN.provisionprofile",
     )
 
+    return [top_app]
+
+
+async def vpn_behavior(config, task, notarize=True):
+    """Notarize vpn app.
+
+    Workflow:
+    . Sign all inner apps
+    . Sign main app
+    . Create pkg
+    . Notarize and staple pkg (optional)
+    . Zip pkg
+    . Move zipped app to artifacts
+
+    Args:
+        config (dict): the running configuration
+        task (dict): the running task
+        notarize (bool): if notarization is enabled
+
+    Raises:
+        IScriptError: on fatal error.
+
+    """
+
+    # Sign the application bundle(s)
+    if "hardened-sign-config" in task["payload"]:
+        top_apps = await sign_hardened_behavior(config, task, create_pkg=False)
+    else:
+        top_apps = await vpn_legacy_behavior(config, task)
```

Comment:
I don't think we need to support the legacy behaviour, IMO it should just use hardened signing + create the pkg the way we need it.

Reply:
Because the `hardened-sign-config` comes from the task payload, there may be a period of time where tasks get submitted both with and without the updated signing method depending on when PRs get rebased to pick up the change. Therefore I felt it a little smarter to keep support for the legacy behavior.